src/Interpreters/ActionsDAG.cpp

```diff
@@ -1923,4 +1924,34 @@ ActionsDAGPtr ActionsDAG::cloneActionsForFilterPushDown(
     return actions;
 }
 
+bool ActionsDAG::isSortingPreserved() const
```

Comment:
This property is a little bit strange.
It may be reasonable to make it dependent on some inputs and return outputs.

It is not very often that function is globally monotonic.
Even if all the functions are globally monotonic, we can remove a sorting key column from result.